### PR TITLE
Prevent goroutine leak from persistent http connections

### PR DESCRIPTION
**Pull request by aaronknister**
###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by jsternberg** [comment]
@aaronknister please make sure to sign the CLA and update the changelog. Thanks.

**Comment by aaronknister** [comment]
Thanks @jsternberg. I've signed the CLA and updated the changelog.

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 ### Bugfixes
 
+- [#6413](https://github.com/influxdata/influxdb/pull/6413): Prevent goroutine leak from persistent http connections.
```
Comment (jsternberg) [inline CHANGELOG.md]: Add it to the bottom and also include a `Thanks @aaronnister.` to the end so you get credit for your work.

Comment (aaronknister) [inline CHANGELOG.md]: 10-4. Done and pushed.

**Pull request merged**